kale/embed/deepdta.py

```diff
@@ -0,0 +1,103 @@
+"""
+DeepDTA based models for drug-target interaction prediction problem.
+"""
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch_geometric.nn import GCNConv
+from torch_geometric.nn import global_max_pool as gmp
+
+
+class DeepDTAEncoder(nn.Module):
+    r"""
+    The DeepDTA's CNNs encoder module, which comprises three 1D-convolutional layers and one max-pooling layer. The module
+    is applied to encoding drug/target sequence information, and the input should be transformed information with
+    integer/label encoding  The original paper is `"DeepDTA: deep drug–target binding affinity prediction"
+    <https://academic.oup.com/bioinformatics/article/34/17/i821/5093245>`_ .
+
+    Args:
+        num_embeddings (int): Number of embedding labels/categories, depends on the types of encoding sequence.
+        embedding_dim (int): Dimensions of embedding labels/categories.
+        sequence_length (int): Max length of input sequence.
+        num_kernels (int): Number of kernels (filters).
+        kernel_length (int): Length of kernel (filter).
+    """
+
+    def __init__(self, num_embeddings, embedding_dim, sequence_length, num_kernels, kernel_length):
+        super(DeepDTAEncoder, self).__init__()
+        self.embedding = nn.Embedding(num_embeddings + 1, embedding_dim)
+        self.conv1 = nn.Conv1d(in_channels=sequence_length, out_channels=num_kernels, kernel_size=kernel_length)
+        self.conv2 = nn.Conv1d(in_channels=num_kernels, out_channels=num_kernels * 2, kernel_size=kernel_length)
+        self.conv3 = nn.Conv1d(in_channels=num_kernels * 2, out_channels=num_kernels * 3, kernel_size=kernel_length)
+        self.gmp = nn.AdaptiveMaxPool1d(output_size=1)
+
+    def forward(self, x):
+        x = self.embedding(x)
+        x = F.relu(self.conv1(x))
+        x = F.relu(self.conv2(x))
+        x = F.relu(self.conv3(x))
+        x = self.gmp(x)
+        x = x.squeeze(2)
+        return x
+
+
+class DrugGCNEncoder(nn.Module):
+    r"""
+    The GraphDTA's GCNs encoder module, which comprises three graph convolutional layers and one full connected layer.
+    The model is a variant of DeepDTA and is applied to encode drug molecule graph information. the original paper
```

Comment:
"applied to encode": be consistent, the above uses "applied to encoding" (after googling, the above may be better, applied to a problem, not an action).

the original paper --> The original paper